onnxscript/rewriter/pattern.py

```diff
@@ -1171,7 +1185,22 @@ def rewrite(cls, op, x: ir.Value, perm: ir.Attr | None = None):
     assert hasattr(rule_class, "pattern"), f"Method 'pattern' is missing from {rule_class!r}."
     assert hasattr(rule_class, "rewrite"), f"Method 'rewrite' is missing from {rule_class!r}."
     assert hasattr(rule_class, "check"), f"Method 'check' is missing from {rule_class!r}."
-    return RewriteRule(rule_class.pattern, rule_class.rewrite, rule_class.check)
+    if generic:
+        import onnxscript.rewriter.generic_pattern as orpp
+
+        return RewriteRule(
+            rule_class.pattern,
+            rule_class.rewrite,
+            rule_class.check,
+            orpp.GenericPatternMatcher,
+            name=rule_class.__name__,  # type: ignore[union-attr]
```

Comment:
I am concerned about breaking this into different paths again.

Reply:
why?